### PR TITLE
Linchpin AppImage on release

### LINKED ISSUE #1114: RFE: Linchpin AppImage (Preview)
Linchpin distribution is currently done only with pip and containers. Both options are not simple and require knowledge and experience. AppImage is a single binary that contains all the required files to run a program, the only downside is it requires installing fuse on the system host (available on most distrobutions) and the file size. However it could be a good alternative in some cases such as getting started.

### PR DESCRIPTION
**Pull request by abraverm**
fixes #1114

### CODE REVIEW

**Comment by abraverm** [comment]
@samvarankashyap @14rcole, the building of App image is really simple but I have no idea how to test the CI changes, suggestions?

File: config/Dockerfiles/JenkinsfileRelease [hunk 1]
```diff
@@ -210,19 +210,26 @@ EOF
                             }
                         }
                     }
+                    stage('release-appimage') {
+                      dir(repoName) {
+                        pipelineUtils.executeInContainer(currentStage, "fedora29", "curl -O https://raw.githubusercontent.com/AppImage/pkg2appimage/master/pkg2appimage")
+                        pipelineUtils.executeInContainer(currentStage, "fedira29", "bash -ex pkg2appimage appimage.yml")
```
Comment (14rcole) [inline config/Dockerfiles/JenkinsfileRelease]: You've got a spelling error on this line - `fedira29`

**Comment by 14rcole** [comment]
@abraverm to answer your question, the only way I really know how to test this is to do a release.  Maybe @skatlapa or @samvarankashyap will have better ideas

File: config/Dockerfiles/release-notes.py [hunk 2]
```diff
@@ -129,5 +129,9 @@ def format_body(tasks):
 release_body = format_body(tasks)
 release_name = milestone
 
-repo.create_git_release(milestone, release_name, release_body,
-                        target_commitish="master")
+release = repo.create_git_release(milestone, release_name, release_body,
```
Comment (skatlapa) [inline config/Dockerfiles/release-notes.py]: @abraverm I am not familiar with appImage, but looking at the script part of the yaml, it looks good to me. If we don't want to risk it we could either maintain different/seperate Docker Image just for appImage release initially. 

The release pipeline itself can be tested locally in contra-env-setup with the current Jenkinsfile if I see it right.

**Comment by 14rcole** [comment]
@skatlapa is there way to do a mock release when this file changes?  We changed something in a release a couple of months ago and it took a few days to get the release out without any bugs

File: config/Dockerfiles/release-notes.py [hunk 3]
```diff
@@ -129,5 +129,9 @@ def format_body(tasks):
...
+                                  target_commitish="master")
```
Comment (skatlapa) [inline config/Dockerfiles/release-notes.py]: @14rcole we could release it to another test branch apart from master to mimic a mock release

**Comment by abraverm** [comment]
Is it possible to make the CI work for my fork, and I will test the release on it?

**Comment by skatlapa** [comment]
You can, but you'll need an openshift cluster set-up. Did you use this before- https://github.com/CentOS-PaaS-SIG/contra-env-setup?

You could basically create a job out of this Jenkinsfile and adjust these parameters based of your fork 
https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/config/Dockerfiles/JenkinsfileRelease#L16-L20

**Comment by abraverm** [comment]
@skatlapa, I haven't tried contra-env-setup yet, do you have example how to run it to deploy Linchpin CI?

**Comment by samvarankashyap** [comment]
@abraverm is there any place where we can publish the app image like Pypi or does it have to be included as a part of github release file.

**Comment by samvarankashyap** [comment]
@14rcole: Regarding the testing of App image, we can have a mock release job with the commit id and see we if can comment out the parts where we check for the version. However, It again comes down to my previous question, given we create an AppImage where do we publish it? 
For flatpak we have https://flathub.org/  do we have anything like that for AppImage?

**Comment by abraverm** [comment]
https://www.appimagehub.com/
But its not like pypi or flathub where you can run some command and it will get the binary for you. So I think Github releases is the better option.

**Comment by skatlapa** [comment]
@abraverm hey, if this can wait, let's discuss this in tomorrow's dev meeting? I have an idea..

**Comment by samvarankashyap** [comment]
@abraverm needs rebase

**Comment by samvarankashyap** [comment]
[merge]

**Comment by samvarankashyap** [comment]
rebased on develop branch

**Comment by samvarankashyap** [comment]
[merge]

**Comment by samvarankashyap** [comment]
[merge]

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**